infra/modules/database/main.tf

```diff
@@ -0,0 +1,362 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+locals {
+  master_username       = "postgres"
+  app_username          = "app"
+  migrator_username     = "migrator"
+  schema_name           = "app"
+  primary_instance_name = "${var.name}-primary"
+  role_manager_name     = "${var.name}-role-manager"
+  role_manager_package  = "${path.root}/role_manager.zip"
+
+  # The ARN that represents the users accessing the database are of the format: "arn:aws:rds-db:<region>:<account-id>:dbuser:<resource-id>/<database-user-name>""
+  # See https://aws.amazon.com/blogs/database/using-iam-authentication-to-connect-with-pgadmin-amazon-aurora-postgresql-or-amazon-rds-for-postgresql/
+  db_user_arn_prefix = "arn:aws:rds-db:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:dbuser:${aws_rds_cluster.db.cluster_resource_id}"
+}
+
+
+#------------------------#
+# Database Configuration #
+#------------------------#
+
+resource "aws_rds_cluster" "db" {
+  # checkov:skip=CKV2_AWS_27:have concerns about sensitive data in logs; want better way to get this information
+  # checkov:skip=CKV2_AWS_8:TODO add backup selection plan using tags
+
+  # cluster identifier is a unique identifier within the AWS account
+  # https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.CreateInstance.html
+  cluster_identifier = var.name
+
+  engine            = "aurora-postgresql"
+  engine_mode       = "provisioned"
+  database_name     = var.database_name
+  port              = var.port
+  master_username   = local.master_username
+  master_password   = aws_ssm_parameter.random_db_password.value
```

Comment:
I came across this while reviewing: https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-rds-integration-aws-secrets-manager/

Should we try this feature here?

See the example here https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#rdsaurora-managed-master-passwords-via-secrets-manager-default-kms-key

Reply:
yeah i had created that as a ticket before when i first learned about it: https://github.com/navapbc/template-infra/issues/286

ok if we push to a future enhancement?